packages/signers/src/transaction-partial-signer.ts

```diff
@@ -9,6 +9,10 @@ export type TransactionPartialSignerConfig = BaseSignerConfig;
 /** Defines a signer capable of signing transactions. */
 export type TransactionPartialSigner<TAddress extends string = string> = Readonly<{
     address: Address<TAddress>;
+    newSignTransactions(
+        transactions: readonly NewTransaction[],
+        config?: TransactionPartialSignerConfig,
+    ): Promise<readonly SignatureDictionary[]>;
```

Comment:
Should we use the new `SignatureMap` instead of `SignatureDictionary` here?

Reply:
I thought about it, but it's a looser type because `SignatureMap` includes null signatures where `SignatureDictionary` doesn't. And the looser type could cause issues because of how we spread signer signatures into transactions - if a signer added null signatures it would remove existing ones. The naming is confusing though!

Reply:
I'll merge as-is for now, but happy to change this if needed, I don't have a strong view. I don't think `SignatureDictionary` is exported from the module so we can do whatever works best  here.